src/components/editMeasure/EditMeasureNav.tsx

```diff
@@ -23,121 +25,58 @@ const MenuItem = styled.li((props: PropTypes) => [
-    pathname !== `${url}/details/measure-steward` &&
-    pathname !== `${url}/details/measure-description` &&
-    pathname !== `${url}/details/measure-copyright` &&
-    pathname !== `${url}/details/measure-disclaimer` &&
-    pathname !== `${url}/details/measure-rationale` &&
-    pathname !== `${url}/details/measure-guidance` &&
-    pathname !== `${url}/details/measure-definition` &&
-    pathname !== `${url}/details/measure-references` &&
-    pathname !== `${url}/details/measure-clinical-recommendation` &&
-    pathname !== `${url}/details/transmission-format` &&
-    pathname !== `${url}/review-info` &&
-    !pathname.startsWith(`${url}/test-cases`) &&
-    !pathname.startsWith(`${url}/groups`)
-  ) {
-    history.push("/404");
-  }
-  const { updateRouteHandlerState } = routeHandlerStore;
-  const [routeHandlerState, setRouteHandlerState] = useState<RouteHandlerState>(
-    routeHandlerStore.state
-  );
-
-  useEffect(() => {
-    const subscription = routeHandlerStore.subscribe(setRouteHandlerState);
-    return () => {
-      subscription.unsubscribe();
-    };
-  }, []);
-  useEffect(() => {
-    const unblock = history.block(({ pathname }, action) => {
-      if (!routeHandlerState.canTravel) {
-        updateRouteHandlerState({ canTravel: false, pendingRoute: pathname });
-        return false;
-      }
-      unblock();
-    });
-    return unblock;
-  }, [
-    history.block,
-    routeHandlerState.canTravel,
-    routeHandlerState.pendingRoute,
-  ]);
-  const [selected, setSelected] = useState<string>("");
-  useEffect(() => {
-    // groups
-    if (
-      pathname.startsWith(`${url}/groups`) ||
-      pathname.startsWith(`${url}/supplemental-data`) ||
-      pathname.startsWith(`${url}/risk-adjustment`) ||
-      pathname.startsWith(`${url}/base-configuration`)
-    ) {
-      setSelected(`${url}/groups`);
-    } else if (pathname.startsWith(`${url}/details`)) {
-      setSelected(`${url}/details`);
-    } else if (pathname === `${url}/cql-editor`) {
-      setSelected(`${url}/cql-editor`);
-    } else if (pathname.startsWith(`${url}/test-cases`)) {
-      setSelected(`${url}/test-cases`);
-    } else if (pathname.startsWith(`${url}/review-info`)) {
-      setSelected(`${url}/review-info`);
-    }
-  }, [pathname]);
+  let navigate = useNavigate();
+  const { id } = useParams<{
+    id: string;
+  }>();
   const handleChange = (e, v) => {
-    setSelected(v);
+    const newPath = `/measures/${id}/edit/${v}`;
+    navigate(newPath, { replace: true });
   };
   const qdmNavTo = () => {
-    isQDM ? `${url}/base-configuration` : `${url}/groups/1`;
+    isQDM ? `${pathname}/base-configuration` : `${pathname}/groups/1`;
   };
-
+  // we grab the matching pattern after edit, then we only get the part before the next slash.
+  const fullMatch = useMatch("/measures/:id/edit/*")?.params?.["*"];
+  if (fullMatch === "details" && id) {
+    navigate(`/measures/${id}/edit/details/`, { replace: true });
```

Comment:
Looks like while navigating replace true is set for all navigations, I might be mis-understanding the functionality of replace option. Let me know if thats the case

Reply:
No, it definitely shouldn't be used here and just breaks back routing.